optuna/integration/sklearn.py

```diff
@@ -130,6 +133,44 @@ def _safe_indexing(
     return sklearn_safe_indexing(X, indices)
 
 
+# NOTE Original implementation:
+# https://github.com/scikit-learn/scikit-learn/blob/ \
+# 28ef9973362257e0627bd39db9b788c93f49362f/sklearn/model_selection/_search.py#L348-L355
+# NOTE This function is newly introduced on Scikit-Learn 1.0.0 (rename of 0.25)
+# TODO(tsuga) Remove this code block and import this function from sklearn after we
+# migrate our dependencies beyond Scikit-Learn 1.0.0
```

Comment:
Time to Say Goodbye. Python 3.6 is end-of-life by 2021/12/23.
My question is, will optuna continue to support python 3.6 (if so, your comment remains valid), or to drop it (if so, I will remove the code block and import the equivalent function from sklearn)?

Reply:
Great! As in https://github.com/optuna/optuna/issues/3021, Optuna slowly drops python 3.6 support. So it is supposed to be a reasonable time to drop python 3.6 for this Scikit-learn integration. 

Note that we need to add 
`--ignore tests/integration_tests/test_sklearn.py` around  
 https://github.com/optuna/optuna/blob/57fc617a1a0f7c94b0f39ae63f47ee2a7f085462/.github/workflows/tests-integration.yml#L61-L65 because Optuna still uses Python 3.6 for other integrations and optuna itself.